controller/controller.go

```diff
@@ -0,0 +1,90 @@
+package controller
+
+import (
+	"dashboard/model"
+	"database/sql"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+var db *sql.DB
+var err error
+
+//HandleSignUp func
+func HandleSignUp(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var user model.User
+	var query string
+	var res model.ResponseResult
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	err = json.Unmarshal(body, &user)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	stmtOut, err := db.Prepare("SELECT username FROM users WHERE username  = ? ")
+	if err != nil {
+		panic(err.Error())
+	}
+	defer stmtOut.Close()
+	err = stmtOut.QueryRow(user.Username).Scan(&query)
+	if err != nil {
+		stmt, err := db.Prepare("INSERT INTO users(Username,FirstName,LastName,Password,Gender,Country,Age,Email) VALUES(?)")
+		if err != nil {
+			log.Panic(err)
+		}
+		defer stmt.Close()
+
+		_, err = stmt.Exec(user.Username, user.FirstName, user.LastName, user.Password, user.Gender, user.Country, user.Age, user.Email)

```

Comment:
do not insert password as plain text in db. Use sha1 library to calculate hash and then store hash in db.
and while sign in again calculate sha1 and match from db.